flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/nodes/physical/batch/BatchExecSink.scala

```diff
@@ -36,10 +46,99 @@ class BatchExecSink[T](
     sink: TableSink[T],
     sinkName: String)
   extends Sink(cluster, traitSet, inputRel, sink, sinkName)
-  with BatchPhysicalRel {
+  with BatchPhysicalRel
+  with BatchExecNode[Any] {
 
   override def copy(traitSet: RelTraitSet, inputs: util.List[RelNode]): RelNode = {
     new BatchExecSink(cluster, traitSet, inputs.get(0), sink, sinkName)
   }
 
+  /**
+    * Returns [[DamBehavior]] of this node.
+    */
+  override def getDamBehavior: DamBehavior = DamBehavior.PIPELINED
+
+  /**
+    * Returns an array of this node's inputs. If there are no inputs,
+    * returns an empty list, not null.
+    *
+    * @return Array of this node's inputs
+    */
+  override def getInputNodes: util.List[ExecNode[BatchTableEnvironment, _]] = {
+    List(getInput.asInstanceOf[ExecNode[BatchTableEnvironment, _]])
+  }
+
+  /**
+    * Internal method, translates this node into a Flink operator.
+    *
+    * @param tableEnv The [[BatchTableEnvironment]] of the translated Table.
+    */
+  override protected def translateToPlanInternal(
+    tableEnv: BatchTableEnvironment): StreamTransformation[Any] = {
+    val convertTransformation = sink match {
+      case _: BatchTableSink[T] =>
+        translate(withChangeFlag = false, tableEnv)
+      case streamTableSink: DataStreamTableSink[T] =>
+        translate(withChangeFlag = streamTableSink.withChangeFlag, tableEnv)
+      case _ =>
+        throw new TableException("Only Support BatchTableSink or BatchCompatibleStreamTableSink")
```

Comment:
BatchCompatibleStreamTableSink could be added when we really need it.

Reply:
I will update error info here.

Reply:
`BatchCompatibleStreamTableSink` caused a few bugs in our internal version, i think we should have a better solution for it. For now, we'd better try to avoid introducing it.